youki_integration_test/src/utils/support.rs

```diff
@@ -21,3 +28,41 @@ pub fn get_project_path() -> PathBuf {
         Err(e) => panic!("directory is not found, {}", e),
     }
 }
+
+/// This will generate the UUID needed when creating the container.
+pub fn generate_uuid() -> Uuid {
+    let mut rng = rand::thread_rng();
+    const CHARSET: &[u8] = b"0123456789abcdefABCDEF";
+
+    let rand_string: String = (0..32)
+        .map(|_| {
+            let idx = rng.gen_range(0..CHARSET.len());
+            CHARSET[idx] as char
+        })
+        .collect();
+
+    match Uuid::parse_str(&rand_string) {
+        Ok(uuid) => uuid,
+        Err(e) => panic!("can not parse uuid, {}", e),
+    }
+}
+
+/// Creates a bundle directory in a temp directory
+pub fn prepare_bundle(id: &Uuid) -> Result<TempDir> {
+    let temp_dir = create_temp_dir(id)?;
+    let tar_file_name = "bundle.tar.gz";
+    let tar_path = std::env::current_dir()?.join(tar_file_name);
+    std::fs::copy(tar_path.clone(), (&temp_dir).join(tar_file_name))?;
+    let tar_gz = File::open(tar_path)?;
+    let tar = GzDecoder::new(tar_gz);
+    let mut archive = Archive::new(tar);
+    archive.unpack(&temp_dir)?;
+    Ok(temp_dir)
+}
+
+/// Sets the config.json file as per given spec
+pub fn set_config<P: AsRef<Path>>(project_path: P, config: &Spec) -> Result<()> {
+    let path = project_path.as_ref().join("bundle").join("config.json");
```

Comment:
Why we can use `oci-spec-rs` builder to dynamically create our own config.json instead of using the `config.json`? In other words, could we generate a `config.json` that could be easily configured programmatically for each test?

Reply:
If I understand your concern correctly, I am doing exactly that here : in tlb_test.rs:13 there is a function which generates the spec using oci-spec-rs builder.
The function set_config is used to save that generated config in the directory on disk, so runtime can use it when creating the container.
If you mean that we should add an interface over oci-spec-rs which will generate a single spec, and modify it as required, I don't think that the editing the config that way would be good, given that the builder already does something similar.
In case I have misunderstood something, can you please once clarify it? :sweat_smile:

Reply:
Oops, I missed that! It's perfect.